### PR TITLE
*: disable v2 API by default

### PR DESCRIPTION
**Pull request by gyuho**
Still need changelog.

/cc @xiang90 @jpbetz @philips

### CODE REVIEW

**Comment by philips** [comment]
The motivation for this is to disable it by default in v3.4?

**Comment by gyuho** [comment]
@philips Yes. We will still keep v2 storage and v2 API, until we have v4.

**Comment by philips** [comment]
OK, this will need major documentation, changelog, and user outreach. Has any of that been planned on an issue?

**Comment by gyuho** [comment]
Yeah I will make sure we highlight this in our documentation as a breaking change.

**Comment by jingyih** [comment]
FYI. Noticed some broken unit test:

https://travis-ci.com/etcd-io/etcd/jobs/219715780#L585

**Comment by gyuho** [comment]
I checked the code base, and confirmed that publish method won't be affected by this. It goes directly to rafthttp. So even with v2 client handler disabled (via `--enable-v2=false`), cluster can still process `pubish` requests (which encodes requests using v2 storage). Added a TODO to replace this later with v3 storage.

File: etcdserver/server.go [hunk 1]
```diff
@@ -1980,6 +1980,12 @@ func (s *EtcdServer) sync(timeout time.Duration) {
 // static clientURLs of the server.
 // The function keeps attempting to register until it succeeds,
 // or its server is stopped.
+//
+// Use v2 store to encode member attributes, and apply through Raft
```
Comment (jingyih) [inline etcdserver/server.go]: Agreed.

**Comment by gyuho** [comment]
@jingyih Now CI should be passing. PTAL. Thx.

**Comment by jingyih** [comment]
LGTM

**Comment by jingyih** [comment]
@gyuho I did not see a corresponding changelog for this?

**Comment by gyuho** [comment]
@jingyih Ah, totally missed that. Let me add now. Thanks.

**Pull request merged**